test/common/testlib.py

```diff
@@ -860,11 +866,22 @@ def check_journal_messages(self, machine=None):
             self.copy_cores("FAIL")
             raise Error(first)
 
-    def check_browser_messages(self):
+    def allow_browser_errors(self, *patterns):
+        """Don't fail if the test caused a console error matching the given regexp"""
+        for p in patterns:
+            self.allowed_console_errors.append(p)
+
+    def check_browser_errors(self):
         if not self.browser:
             return
         for log in self.browser.get_js_log():
-            if log.startswith("error: uncaught"):
+            if not log.startswith("error: "):
+                continue
+            # errors are fatal in general; they need to be explicitly whitelisted
+            for p in self.allowed_console_errors:
+                if re.search(p, log):
```

Comment:
Good point. Honestly I find that journal message patterns should use "match" as well -- a lot of  them just end up with a `.*` prefix or suffix, and I haven't seen realistic cases where a match instead of search would introduce ambiguity.

When moving console errors to full matching, one needs to take care to  always include the `error: ` prefix; not a big deal either, though, given how rare they are.

So for this PR I propose to just add comments to their initial documentation, and docstrings to the `allow_()` functions. A follow-up PR could then reconcile this by moving the journal patterns to "match", and drop all the extra `.*`. Does that work for you?

Reply:
Pushed again with [this diff](https://paste.fedoraproject.org/paste/uJhSGkLMBlOlFH6F6YsrAQ) to document match vs. search, and changed the commit message accordingly.